contracts/MemeFactory.sol

```diff
@@ -286,10 +289,79 @@ abstract contract MemeFactory {
         emit Collected(msg.sender, memeToken, allocation);
     }
 
+    /// @dev Create a new meme token.
+    function _createThisMeme(
+        uint256 memeNonce,
+        string memory name,
+        string memory symbol,
+        uint256 totalSupply
+    ) internal returns (address memeToken) {
+        bytes32 randomNonce = keccak256(abi.encodePacked(block.timestamp, msg.sender, memeNonce));
+        randomNonce = keccak256(abi.encodePacked(randomNonce));
+        bytes memory payload = abi.encodePacked(type(Meme).creationCode, abi.encode(name, symbol, DECIMALS, totalSupply));
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            memeToken := create2(0x0, add(0x20, payload), mload(payload), memeNonce) // revert to randomNonce
+        }
+
+        // Check for non-zero token address
+        require(memeToken != address(0), "Token creation failed");
+    }
+
     /// @dev Allows diverting first x collected funds to a launch campaign.
     /// @return adjustedAmount Adjusted amount of native token to convert to OLAS and burn.
     function _launchCampaign() internal virtual returns (uint256 adjustedAmount);
 
+    /// @dev Unleashes the meme token.
```

Comment:
This was already checked when shuffling functions, somehow the reference PR does not point to the correct one.